exilez/init/code/ZombieDeleter.sqf

```diff
@@ -1,39 +1,43 @@
 // ExileZ 2.0 by Patrix87 of http:\\multi-jeux.quebec //
 
-private ["_device","_zombie","_zombiePos","_zombieClass","_distanceDeath","_distance","_radius","_avoidTerritory","_flags"];
+private ["_device","_zombie","_zombiePos","_zombieClass","_distanceDeath","_distance","_radius",/*"_avoidTerritory",*/"_flags"];
 
 _zombie = _this select 0;
-_avoidTerritory = _this select 1;
+//_avoidTerritory = _this select 1;
 _zombieClass = typeOf _zombie;
 _distanceDeath = false;
 
 while {alive _zombie} do {
 	sleep MaxTime;
 	_zombiePos = getPos _zombie;
+	
 	//check for the absence of players
 	if (({isplayer _x} count (_zombiePos nearEntities MaxDistance) == 0) && alive _zombie) then {
 		_zombie setdamage 1;
 		sleep 5;
 		deleteVehicle _zombie;
 		_distanceDeath = true;
 	};
+	
+	//check for safe zones
+	if ((RemoveZfromTraders) && ((getPosATL _zombie) call ExileClient_util_world_isInTraderZone) && (alive _zombie)) then
+	{
+		_zombie setdamage 1;
+		sleep 5;
+		deleteVehicle _zombie;
+		_distanceDeath = true;
+	};
+	
 	//check for flags
```

Comment:
Ok that is the part I was talking about when I said your are removing a feature.

_avoidTerritory is a flag you can add to your trigger group config so that the zombies can or cannot go inside the territories if RemoveZfromTerritory is false.

Some server admin might want that.

in fact there's an error in my original code
`if (RemoveZfromTerritory && _avoidTerritory && alive _zombie)then`
should be 
`if ((RemoveZfromTerritory || _avoidTerritory) && alive _zombie)then`

and inside that if, add another if to check if the zombie is in the territory
```
if ((getPosATL _zombie) call ExileClient_util_world_isInTerritory)then
{
    _zombie setdamage 1;
    sleep 5;
    deleteVehicle _zombie;
    _distanceDeath = true;
}
```
then I'll merge your code, everything else is fine.